mm2src/lp_ordermatch.rs

```diff
@@ -2716,46 +2722,76 @@ pub async fn clean_memory_loop(ctx: MmArc) {
 
 async fn process_maker_reserved(ctx: MmArc, from_pubkey: H256Json, reserved_msg: MakerReserved) {
     let ordermatch_ctx = OrdermatchContext::from_ctx(&ctx).unwrap();
+    {
+        let my_taker_orders = ordermatch_ctx.my_taker_orders.lock().await;
+        if !my_taker_orders.contains_key(&reserved_msg.taker_order_uuid) {
+            return;
+        }
+    }
+
     let our_public_id = ctx.public_id().unwrap();
     if our_public_id.bytes == from_pubkey.0 {
         log::warn!("Skip maker reserved from our pubkey");
         return;
     }
 
+    let uuid = reserved_msg.taker_order_uuid;
+    {
+        let mut pending_map = ordermatch_ctx.pending_maker_reserved.lock().await;
+        let pending_for_order = pending_map
+            .entry(reserved_msg.taker_order_uuid)
+            .or_insert_with(Vec::new);
+        pending_for_order.push(reserved_msg);
+        if pending_for_order.len() > 1 {
+            // messages will be sorted by price and processed in the first called handler
+            return;
+        }
+    }
+
+    Timer::sleep(3.).await;
```

Comment:
Just one question to make sure I understand correctly. is this `Timer::sleep` used here to wait for other MakerReserved messages to be processed so I can have all the orders that matched with my TakerRequest before sorting by price?

Reply:
Yes, that's correct. The responsibility to match the order is left to the first called handler. All others just add a new pending match and return immediately.